src/doom/d_main.c

```diff
@@ -1094,7 +1094,7 @@ static void InitGameVersion(void)
                 gameversion = exe_doom_1_1;
 
                 // detect v1.0 from length of demo1 lump
-                if (demo1 && W_LumpLength(demo1) < 10 * 1024)
+                if (demo1 > -1 && W_LumpLength(demo1) < 10 * 1024)
```

Comment:
As the checks in `InitGameVersion()` become more and more complex, we should probably consider moving all this away from `doom/d_main.c` into a separate file?

Reply:
Yes, +1 on this. This is my main beef with this PR as it currently stands.

The changes necessary to support v1.0 can be broadly categorized as follows:

1. Patch some of the normal tables and constants to match v1.0.
2. Change some of the game logic to match v1.0 (with `if (gameversion...) {...}`

The problem with (2) is that it can introduce a lot of complexity to the code if we're not careful. So we should try to avoid doing this as much as possible. What can be rearchitected here to put things into the (1) category instead of (2)? Try to get as few new `if` statements as possible.

My immediate suggestions for how to proceed on this PR are as follows:

* Make a new file - maybe named something like d_oldversion.c - and try to isolate as much as possible to that file. That includes the "Version-specific differences" switch statement below, and the alternative versions of the trig tables, so we move those out of `tables.c`.

* It should be possible to patch the `forwardmove[]` and `sidemove[]` tables from the new file too, so we avoid adding that complexity.

* Define new functions named eg. `D_CoarseSine` and `D_CoarseCosine` that will either look up into the `coarsesine[]` table or the `finesine[]` table depending on the version.

Reply:
Done in https://github.com/chocolate-doom/chocolate-doom/pull/1207/commits/d8f6bcf685da5f096480d21cc0162de06d1b6f24 .